pandas/tests/frame/test_analytics.py

```diff
@@ -23,54 +23,47 @@
 
 import pandas.util.testing as tm
 import pandas.util._test_decorators as td
-from pandas.tests.frame.common import TestData
 
 
-class TestDataFrameAnalytics(TestData):
+class TestDataFrameAnalytics():
 
     # ---------------------------------------------------------------------=
     # Correlation and covariance
 
     @td.skip_if_no_scipy
-    def test_corr_pearson(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_pearson(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('pearson')
+        self._check_method(float_frame, 'pearson')
 
     @td.skip_if_no_scipy
-    def test_corr_kendall(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_kendall(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('kendall')
+        self._check_method(float_frame, 'kendall')
 
     @td.skip_if_no_scipy
-    def test_corr_spearman(self):
-        self.frame['A'][:5] = nan
-        self.frame['B'][5:10] = nan
+    def test_corr_spearman(self, float_frame):
+        float_frame['A'][:5] = nan
+        float_frame['B'][5:10] = nan
 
-        self._check_method('spearman')
+        self._check_method(float_frame, 'spearman')
 
-    def _check_method(self, method='pearson', check_minp=False):
-        if not check_minp:
-            correls = self.frame.corr(method=method)
-            exp = self.frame['A'].corr(self.frame['C'], method=method)
-            tm.assert_almost_equal(correls['A']['C'], exp)
-        else:
-            result = self.frame.corr(min_periods=len(self.frame) - 8)
-            expected = self.frame.corr()
-            expected.loc['A', 'B'] = expected.loc['B', 'A'] = nan
-            tm.assert_frame_equal(result, expected)
+    def _check_method(self, frame, method='pearson'):
+        correls = frame.corr(method=method)
+        exp = frame['A'].corr(frame['C'], method=method)
+        tm.assert_almost_equal(correls['A']['C'], exp)
```

Comment:
Here you can use `result` and `expected` in spite of the existing code using `exp`

Reply:
Um, first you ask me to remove all orthogonal changes, and then you ask me to add... orthogonal changes? Same happened with the import clean-up other PRs, but when I preempted them here, you ask me to remove them - can you understand how exasperating this back-and-forth is?